transactions/src/main/java/org/jboss/as/txn/service/JTAEnvironmentBeanService.java

```diff
@@ -56,7 +57,9 @@ public void start(StartContext context) throws StartException {
         // recovery nodes
         jtaEnvironmentBean.setXaRecoveryNodes(Collections.singletonList(nodeIdentifier));
         // setup the XA orphan filters
-        jtaEnvironmentBean.setXaResourceOrphanFilterClassNames(Arrays.asList(JTATransactionLogXAResourceOrphanFilter.class.getName(), JTANodeNameXAResourceOrphanFilter.class.getName(), SubordinateJTAXAResourceOrphanFilter.class.getName()));
+        if (System.getProperty("JTAEnvironmentBean.xaResourceOrphanFilterClassNames") == null) {
```

Comment:
The narayana environment bean initialization code already consults environment variables so this check should not be required. Instead I would expect the following:

    if (cnames.size() == 0)
        jtaEnvironmentBean.setXaResourceOrphanFilterClassNames(etc)